src/mono/cmake/configure.cmake

```diff
@@ -175,3 +175,9 @@ if(HOST_SOLARIS)
   set(HAVE_NETINET_TCP_H 1)
   set(HAVE_GETADDRINFO 1)
 endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+  # FreeBSD fails earlier ac_check_headers for these because <sys/types.h> is needed.
```

Comment:
I'm wondering if we should fix the detection at some point..

Reply:
the `ac_check_headers` for the items that we must manually define will work fine for Linux systems which don't need `<sys/types.h>`.

Reply:
We could add them even for Linux, right? I think we did it in few other places. But I don't see it critical - just may be easier to maintain.

Reply:
I tried a few different ways to do that but I'm new at CMake and couldn't get it working here without brute force. It's probably something simple I missed.

Reply:
On the coreclr side, `sysctl` is explictly marked as depreciated on Linux via comment and `set(HAVE_SYS_SYSCTL_H 0)` via a `if (CLR_CMAKE_TARGET_LINUX)` (https://github.com/dotnet/runtime/blob/214443193a83b47170a8bec7ab4247ca745ccc8b/src/libraries/Native/Unix/configure.cmake#L904-L#911)

Reply:
I'd need someone to suggest a change here - I don't have a different solution.

Reply:
honestly? leave it as is. most of the `ac_check_headers` and `ac_check_funcs` should be reworked so that all these other platforms don't need to make their own hardcoded settings, but that might be quite a challenge. 

If you really want to change the detection you can do what was done in the `src/libraries/Native/Unix/configure.cmake` (linked above)

Reply:
I'm not sure the solution in the link is any more simple/clean. I'm inclined to leave it for now and let someone with a stronger conviction take a go at it later.